tests/phpunit/api/v3/SettingTest.php

```diff
@@ -546,4 +546,40 @@ public function testDefaults() {
     $this->assertEquals('Unconfirmed', $result['values'][$dom['id']]['tag_unconfirmed']);
   }
 
+  /**
+   * Test to set civicrmEnvironment
+   *
+   */
+  public function testSetCivicrmEnvironment() {
+    $params = array(
+      'civicrmEnvironment' => TRUE,
+    );
+    $result = $this->callAPISuccess('Setting', 'create', $params);
+    $params = array(
+      'name' => 'civicrmEnvironment',
+      'group' => 'Developer Preferences',
+    );
+    $result = $this->callAPISuccess('Setting', 'getvalue', $params);
+    $this->assertEquals(TRUE, $result);
+
+    global $civicrm_setting;
+    $civicrm_setting[CRM_Core_BAO_Setting::DEVELOPER_PREFERENCES_NAME]['civicrmEnvironment'] = FALSE;
+    Civi::service('settings_manager')->useMandatory();
+    $result = $this->callAPISuccess('Setting', 'getvalue', $params);
+    $this->assertEquals(FALSE, $result);
+
+    // check for failure
```

Comment:
// check for failure if trying to set value to one different than what is set in civicrm.settings.php

change code to a procedure similar to outlined above.